Build failure: givaro (dependency of sage)

## Steps To Reproduce

Steps to reproduce the behavior:

1. build givaro

## Build log


<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/0kv7qpscrhlcm50zp52z14wmdriv5a3f-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/mkdr6cinf89haa07jbsm35r0f7jjhz74-gcc-13.patch
patching file src/library/poly1/givdegree.h
applying patch /nix/store/1q2vk5ka8sq46dazpl632c4vk95432zg-clang-16.patch
patching file src/kernel/integer/random-integer.h
Running phase: autoreconfPhase
@nix { "action": "setPhase", "phase": "autoreconfPhase" }
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I macros
autoreconf: configure.ac: tracing
autoreconf: configure.ac: creating directory build-aux
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'macros'.
libtoolize: copying file 'macros/libtool.m4'
libtoolize: copying file 'macros/ltoptions.m4'
libtoolize: copying file 'macros/ltsugar.m4'
libtoolize: copying file 'macros/ltversion.m4'
libtoolize: copying file 'macros/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I macros
autoreconf: running: /nix/store/m0knd99ncm7kbpffm10gsdmgv1j6mgy6-autoconf-2.72/bin/autoconf --force
configure.ac:35: warning: The macro 'AC_PROG_LIBTOOL' is obsolete.
configure.ac:35: You should run autoupdate.
macros/libtool.m4:100: AC_PROG_LIBTOOL is expanded from...
configure.ac:35: the top level
configure.ac:40: warning: The macro 'AC_HEADER_STDC' is obsolete.
configure.ac:40: You should run autoupdate.
./lib/autoconf/headers.m4:663: AC_HEADER_STDC is expanded from...
configure.ac:40: the top level
configure.ac:47: warning: The macro 'AC_TRY_RUN' is obsolete.
configure.ac:47: You should run autoupdate.
./lib/autoconf/general.m4:2995: AC_TRY_RUN is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
macros/debug.m4:53: AC_COMPILER_NAME is expanded from...
configure.ac:47: the top level
configure.ac:61: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:61: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1534: AC_ARG_ENABLE is expanded from...
macros/debug.m4:128: AC_INLINE is expanded from...
configure.ac:61: the top level
configure.ac:64: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:64: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1534: AC_ARG_ENABLE is expanded from...
macros/debug.m4:9: AC_DEBUG is expanded from...
macros/common.m4:10: SET_FLAGS is expanded from...
configure.ac:64: the top level
configure.ac:64: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:64: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1534: AC_ARG_ENABLE is expanded from...
macros/debug.m4:22: AC_PROFILE is expanded from...
macros/common.m4:10: SET_FLAGS is expanded from...
configure.ac:64: the top level
configure.ac:64: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:64: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1534: AC_ARG_ENABLE is expanded from...
macros/debug.m4:36: AC_WARNINGS is expanded from...
macros/common.m4:10: SET_FLAGS is expanded from...
configure.ac:64: the top level
configure.ac:69: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:69: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1553: AC_ARG_WITH is expanded from...
macros/common.m4:90: ARCH_FLAGS is expanded from...
configure.ac:69: the top level
configure.ac:86: warning: The macro 'AC_TRY_COMPILE' is obsolete.
configure.ac:86: You should run autoupdate.
./lib/autoconf/general.m4:2845: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/types.m4:149: _AC_CHECK_TYPE_NEW is expanded from...
./lib/autoconf/types.m4:233: AC_CHECK_TYPE is expanded from...
configure.ac:86: the top level
configure.ac:123: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:123: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1553: AC_ARG_WITH is expanded from...
macros/gmp-check.m4:14: GIV_CHECK_GMP is expanded from...
configure.ac:123: the top level
configure.ac:123: warning: The macro 'AC_TRY_RUN' is obsolete.
configure.ac:123: You should run autoupdate.
./lib/autoconf/general.m4:2995: AC_TRY_RUN is expanded from...
macros/gmp-check.m4:14: GIV_CHECK_GMP is expanded from...
configure.ac:123: the top level
configure.ac:126: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:126: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1553: AC_ARG_WITH is expanded from...
macros/givaro-doc.m4:13: GIV_DOC is expanded from...
configure.ac:126: the top level
configure.ac:126: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:126: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1534: AC_ARG_ENABLE is expanded from...
macros/givaro-doc.m4:13: GIV_DOC is expanded from...
configure.ac:126: the top level
autoreconf: running: /nix/store/m0knd99ncm7kbpffm10gsdmgv1j6mgy6-autoconf-2.72/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:32: installing 'build-aux/ar-lib'
configure.ac:32: installing 'build-aux/compile'
configure.ac:17: installing 'build-aux/config.guess'
configure.ac:17: installing 'build-aux/config.sub'
configure.ac:19: installing 'build-aux/install-sh'
configure.ac:19: installing 'build-aux/missing'
benchmarks/Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
autoreconf: 'build-aux/install-sh' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing libtool script ./build-aux/ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/27b7iljyhsjmv3gxicw2m5fl5gyk93dx-givaro-4.2.0 --without-archnative --disable-sse3 --disable-ssse3 --disable-sse41 --disable-sse42 --disable-avx --disable-avx2 --disable-fma --disable-fma4
configure: WARNING: unrecognized options: --disable-sse3, --disable-ssse3, --disable-sse41, --disable-sse42, --disable-avx, --disable-avx2, --disable-fma, --disable-fma4
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking whether make supports the include directive... yes (GNU style)
checking dependency style of g++... none
checking for gcc... gcc
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... none
checking the archiver (ar) interface... ar
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/abm77lnrkrkb58z6xp1qwjcr1xgkcfwm-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd
checking how to truncate binary pipes... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... ld -m elf_x86_64
checking if the linker (ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for egrep... (cached) /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -E
checking for float.h... yes
checking for limits.h... yes
checking for stddef.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for sys/time.h... yes
checking for stdint.h... (cached) yes
checking for pthread.h... yes
## --------------------------------- ##
##        START GIVARO CONFIG        ##
## --------------------------------- ##
checking for family name of compiler... gcc
checking whether g++ supports C++11 features by default... yes
---------------------------------------
checking whether to inline or not most of the code ?... no
checking whether to enable debugging options in the library... no
checking whether to enable profiling everything in the library... no
checking whether to enable warnings when compiling the library... no
checking whether C++ compiler accepts -march=native... yes
---------------------------------------
checking size of char... 1
checking size of short... 2
checking size of int... 4
checking size of long... 8
checking size of long long... 8
checking size of __int64_t... 8
checking for __int128_t... yes
checking whether byte ordering is bigendian... no
configure: Creating symbolic links for compilation
---------------------------------------
checking for gmp.h... yes
checking for GMP library... yes
checking for gmpxx.h... yes
checking for GMP cxx support... yes
checking whether gmp version is at least 40000... yes
checking whether to build documentation... no
---------------------------------------
GIVARO_CXXFLAGS =   -O2 -Wall -DNDEBUG -UDEBUG
## --------------------------------- ##
##         END GIVARO CONFIG         ##
## --------------------------------- ##
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating givaro-makefile
config.status: creating src/Makefile
config.status: creating src/kernel/Makefile
config.status: creating src/kernel/system/Makefile
config.status: creating src/kernel/memory/Makefile
config.status: creating src/kernel/gmp++/Makefile
config.status: creating src/kernel/integer/Makefile
config.status: creating src/kernel/bstruct/Makefile
config.status: creating src/kernel/rational/Makefile
config.status: creating src/kernel/recint/Makefile
config.status: creating src/kernel/field/Makefile
config.status: creating src/kernel/ring/Makefile
config.status: creating src/library/Makefile
config.status: creating src/library/vector/Makefile
config.status: creating src/library/tools/Makefile
config.status: creating src/library/poly1/Makefile
config.status: creating src/library/matrix/Makefile
config.status: creating macros/Makefile
config.status: creating benchmarks/Makefile
config.status: creating examples/Makefile
config.status: creating examples/FiniteField/Makefile
config.status: creating examples/Integer/Makefile
config.status: creating examples/Rational/Makefile
config.status: creating examples/RecInt/Makefile
config.status: creating examples/Polynomial/Makefile
config.status: creating examples/Matrix/Makefile
config.status: creating tests/Makefile
config.status: creating docs/Makefile
config.status: creating givaro.pc
config.status: creating givaro-config
config.status: creating config.h
config.status: executing givaro-config.h commands
config.status: creating givaro-config.h - prefix __GIVARO for config.h defines
config.status: executing depfiles commands
config.status: executing libtool commands
configure: WARNING: unrecognized options: --disable-sse3, --disable-ssse3, --disable-sse41, --disable-sse42, --disable-avx, --disable-avx2, --disable-fma, --disable-fma4
---------------------------------------
configurePhase completed in 42 seconds
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j12 SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
make  all-recursive
make[1]: Entering directory '/build/source'
Making all in src
make[2]: Entering directory '/build/source/src'
Making all in kernel
make[3]: Entering directory '/build/source/src/kernel'
Making all in system
make[4]: Entering directory '/build/source/src/kernel/system'
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../.. -I../../../src/kernel/memory  -O2 -Wall -DNDEBUG -UDEBUG  -c -o givbasictype.lo givbasictype.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../.. -I../../../src/kernel/memory  -O2 -Wall -DNDEBUG -UDEBUG  -c -o giverror.lo giverror.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../.. -I../../../src/kernel/memory  -O2 -Wall -DNDEBUG -UDEBUG  -c -o givinit.lo givinit.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../.. -I../../../src/kernel/memory  -O2 -Wall -DNDEBUG -UDEBUG  -c -o givmodule.lo givmodule.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../.. -I../../../src/kernel/memory  -O2 -Wall -DNDEBUG -UDEBUG  -c -o givtimer.lo givtimer.C
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -O2 -Wall -DNDEBUG -UDEBUG -c givbasictype.C  -fPIC -DPIC -o .libs/givbasictype.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -O2 -Wall -DNDEBUG -UDEBUG -c givtimer.C  -fPIC -DPIC -o .libs/givtimer.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -O2 -Wall -DNDEBUG -UDEBUG -c givinit.C  -fPIC -DPIC -o .libs/givinit.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -O2 -Wall -DNDEBUG -UDEBUG -c giverror.C  -fPIC -DPIC -o .libs/giverror.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -O2 -Wall -DNDEBUG -UDEBUG -c givmodule.C  -fPIC -DPIC -o .libs/givmodule.o
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=link g++ -O2 -Wall -DNDEBUG -UDEBUG    -o libgivsystem.la  givbasictype.lo giverror.lo givinit.lo givmodule.lo givtimer.lo  
libtool: link: ar cr .libs/libgivsystem.a .libs/givbasictype.o .libs/giverror.o .libs/givinit.o .libs/givmodule.o .libs/givtimer.o 
libtool: link: ranlib .libs/libgivsystem.a
libtool: link: ( cd ".libs" && rm -f "libgivsystem.la" && ln -s "../libgivsystem.la" "libgivsystem.la" )
make[4]: Leaving directory '/build/source/src/kernel/system'
Making all in memory
make[4]: Entering directory '/build/source/src/kernel/memory'
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../.. -I../../../src/kernel/system  -O2 -Wall -DNDEBUG -UDEBUG  -c -o givaromm.lo givaromm.C
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/system -O2 -Wall -DNDEBUG -UDEBUG -c givaromm.C  -fPIC -DPIC -o .libs/givaromm.o
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=link g++ -O2 -Wall -DNDEBUG -UDEBUG    -o libgivmemory.la  givaromm.lo  
libtool: link: ar cr .libs/libgivmemory.a .libs/givaromm.o 
libtool: link: ranlib .libs/libgivmemory.a
libtool: link: ( cd ".libs" && rm -f "libgivmemory.la" && ln -s "../libgivmemory.la" "libgivmemory.la" )
make[4]: Leaving directory '/build/source/src/kernel/memory'
Making all in gmp++
make[4]: Entering directory '/build/source/src/kernel/gmp++'
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_lib.lo gmp++_int_lib.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_div.lo gmp++_int_div.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_mod.lo gmp++_int_mod.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_sub.lo gmp++_int_sub.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_compare.lo gmp++_int_compare.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_gcd.lo gmp++_int_gcd.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_io.lo gmp++_int_io.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_mul.lo gmp++_int_mul.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_add.lo gmp++_int_add.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_cstor.lo gmp++_int_cstor.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_misc.lo gmp++_int_misc.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint  -O2 -Wall -DNDEBUG -UDEBUG  -c -o gmp++_int_pow.lo gmp++_int_pow.C
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_div.C  -fPIC -DPIC -o .libs/gmp++_int_div.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_lib.C  -fPIC -DPIC -o .libs/gmp++_int_lib.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_cstor.C  -fPIC -DPIC -o .libs/gmp++_int_cstor.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_misc.C  -fPIC -DPIC -o .libs/gmp++_int_misc.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_add.C  -fPIC -DPIC -o .libs/gmp++_int_add.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_gcd.C  -fPIC -DPIC -o .libs/gmp++_int_gcd.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_mod.C  -fPIC -DPIC -o .libs/gmp++_int_mod.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_sub.C  -fPIC -DPIC -o .libs/gmp++_int_sub.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_pow.C  -fPIC -DPIC -o .libs/gmp++_int_pow.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_compare.C  -fPIC -DPIC -o .libs/gmp++_int_compare.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_io.C  -fPIC -DPIC -o .libs/gmp++_int_io.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel/recint -O2 -Wall -DNDEBUG -UDEBUG -c gmp++_int_mul.C  -fPIC -DPIC -o .libs/gmp++_int_mul.o
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=link g++ -O2 -Wall -DNDEBUG -UDEBUG    -o libgmppp.la  gmp++_int_lib.lo gmp++_int_div.lo gmp++_int_mod.lo gmp++_int_sub.lo gmp++_int_compare.lo gmp++_int_gcd.lo gmp++_int_io.lo gmp++_int_mul.lo gmp++_int_add.lo gmp++_int_cstor.lo gmp++_int_misc.lo gmp++_int_pow.lo  
libtool: link: ar cr .libs/libgmppp.a .libs/gmp++_int_lib.o .libs/gmp++_int_div.o .libs/gmp++_int_mod.o .libs/gmp++_int_sub.o .libs/gmp++_int_compare.o .libs/gmp++_int_gcd.o .libs/gmp++_int_io.o .libs/gmp++_int_mul.o .libs/gmp++_int_add.o .libs/gmp++_int_cstor.o .libs/gmp++_int_misc.o .libs/gmp++_int_pow.o 
libtool: link: ranlib .libs/libgmppp.a
libtool: link: ( cd ".libs" && rm -f "libgmppp.la" && ln -s "../libgmppp.la" "libgmppp.la" )
make[4]: Leaving directory '/build/source/src/kernel/gmp++'
Making all in integer
make[4]: Entering directory '/build/source/src/kernel/integer'
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel -I../../../src/kernel/ring -I../../../src/kernel/rational -I../../../src/kernel  -O2 -Wall -DNDEBUG -UDEBUG  -c -o givinteger.lo givinteger.C
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../..  -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel -I../../../src/kernel/ring -I../../../src/kernel/rational -I../../../src/kernel  -O2 -Wall -DNDEBUG -UDEBUG  -c -o givintprime.lo givintprime.C
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel -I../../../src/kernel/ring -I../../../src/kernel/rational -I../../../src/kernel -O2 -Wall -DNDEBUG -UDEBUG -c givintprime.C  -fPIC -DPIC -o .libs/givintprime.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../.. -I../../../src/kernel/memory -I../../../src/kernel/system -I../../../src/kernel -I../../../src/kernel/ring -I../../../src/kernel/rational -I../../../src/kernel -O2 -Wall -DNDEBUG -UDEBUG -c givinteger.C  -fPIC -DPIC -o .libs/givinteger.o
In file included from [01m[Kgivaro/givinteger.h:24[m[K,
                 from [01m[Kgivinteger.C:13[m[K:
[01m[Kgivaro/givaro/random-integer.h:[m[K In member function '[01m[KGivaro::RandomIntegerIterator<_Unsigned, _Exact_Size>& Givaro::RandomIntegerIterator<_Unsigned, _Exact_Size>::[01;32m[Koperator=[m[K(const Givaro::RandomIntegerIterator<_Unsigned, _Exact_Size>&)[m[K':
[01m[Kgivaro/givaro/random-integer.h:90:54:[m[K [01;31m[Kerror: [m[Kno match for '[01m[Koperator=[m[K' (operand types are '[01m[KGivaro::RandomIntegerIterator<_Unsigned, _Exact_Size>::Integer_Domain[m[K' {aka '[01m[KGivaro::ZRing<Givaro::Integer>[m[K'} and '[01m[Kconst Givaro::RandomIntegerIterator<_Unsigned, _Exact_Size>::Integer_Domain[m[K' {aka '[01m[Kconst Givaro::ZRing<Givaro::Integer>[m[K'})
   90 |                 const_cast<Integer_Domain&>(_ring)=R.[01;31m[K_ring[m[K;
      |                                                      [01;31m[K^~~~~[m[K
make[4]: *** [Makefile:470: givinteger.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
In file included from [01m[Kgivaro/givaro/givinteger.h:24[m[K,
                 from [01m[Kgivaro/givintprime.h:24[m[K,
                 from [01m[Kgivintprime.C:19[m[K:
[01m[Kgivaro/givaro/givaro/random-integer.h:[m[K In member function '[01m[KGivaro::RandomIntegerIterator<_Unsigned, _Exact_Size>& Givaro::RandomIntegerIterator<_Unsigned, _Exact_Size>::[01;32m[Koperator=[m[K(const Givaro::RandomIntegerIterator<_Unsigned, _Exact_Size>&)[m[K':
[01m[Kgivaro/givaro/givaro/random-integer.h:90:54:[m[K [01;31m[Kerror: [m[Kno match for '[01m[Koperator=[m[K' (operand types are '[01m[KGivaro::RandomIntegerIterator<_Unsigned, _Exact_Size>::Integer_Domain[m[K' {aka '[01m[KGivaro::ZRing<Givaro::Integer>[m[K'} and '[01m[Kconst Givaro::RandomIntegerIterator<_Unsigned, _Exact_Size>::Integer_Domain[m[K' {aka '[01m[Kconst Givaro::ZRing<Givaro::Integer>[m[K'})
   90 |                 const_cast<Integer_Domain&>(_ring)=R.[01;31m[K_ring[m[K;
      |                                                      [01;31m[K^~~~~[m[K
make[4]: *** [Makefile:470: givintprime.lo] Error 1
make[4]: Leaving directory '/build/source/src/kernel/integer'
make[3]: *** [Makefile:386: all-recursive] Error 1
make[3]: Leaving directory '/build/source/src/kernel'
make[2]: *** [Makefile:540: all-recursive] Error 1
make[2]: Leaving directory '/build/source/src'
make[1]: *** [Makefile:606: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:434: all] Error 2
```

</details>

## Additional context

This is a result of rebuilding my system with unstable branch, with an overlay overriding sage to the master version, which should have givaro on master. I've heard ecm to break with newer gcc versions, maybe this is a similar break?

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.66, NixOS, 25.05 (Warbler), 25.05.20241217.4989a24`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/9bdl6wa2qck3q0f7l892plk5lwy3pvx1-source`
```

This nixpkgs does not represent the one with sage.

The one should be like this (time/date is UTF+8):
```
├───nixpkgs: github:nixos/nixpkgs/0e82ab234249d8eee3e8c91437802b32c74bb3fd?narHash=sha256-zExSA1i/b%2B1NMRhGGLtNfFGXgLtgo%2BdcuzHzaWA6w3Q%3D (2024-12-30 10:25:05)
├───nixpkgs-master: github:nixos/nixpkgs/b1162df0ae0e53160eb63354b83f2ec68ddcc60e?narHash=sha256-nD%2B9nMwGdISpFvoit3mc/WEIIVhUeUqZN4M9s0bdXQk%3D (2025-01-01 05:06:06)
```

## Notify maintainers

@7c6f434c 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/givaro/default.nix
+++ b/pkgs/development/libraries/givaro/default.nix
@@ -31,6 +31,13 @@ stdenv.mkDerivation rec {
       url = "https://github.com/linbox-team/givaro/commit/a81d44b3b57c275bcb04ab00db79be02561deaa2.patch";
       hash = "sha256-sSk+VWffoEjZRTJcHRISLHPyW6yuvI1u8knBOfxNUIE=";
     })
+    # https://github.com/linbox-team/givaro/issues/226
+    (fetchpatch {
+      name = "gcc-14.patch";
+      url = "https://github.com/linbox-team/givaro/commit/b0cf33e1d4437530c7e4b3db90b6c80057a7f2f3.patch";
+      includes = [ "src/kernel/integer/random-integer.h" ];
+      hash = "sha256-b2Q8apP9ueEqIUtibTeP47x6TlroRzLgAxuv5ZM1EUw=";
+    })
   ];
 
   enableParallelBuilding = true;

--- a/pkgs/development/libraries/givaro/default.nix
+++ b/pkgs/development/libraries/givaro/default.nix
@@ -38,6 +38,12 @@ stdenv.mkDerivation rec {
       includes = [ "src/kernel/integer/random-integer.h" ];
       hash = "sha256-b2Q8apP9ueEqIUtibTeP47x6TlroRzLgAxuv5ZM1EUw=";
     })
+    # https://github.com/linbox-team/givaro/issues/232
+    (fetchpatch {
+      name = "clang-19.patch";
+      url = "https://github.com/linbox-team/givaro/commit/a18baf5227d4f3e81a50850fe98e0d954eaa3ddb.patch";
+      hash = "sha256-IR0IHhCqbxgtsST30vxM9ak1nGtt0apxcLUQ1kS1DHw=";
+    })
   ];
 
   enableParallelBuilding = true;

